NPE: TemporalAdapter.COMPARATOR is null

**Describe the bug**

When initializing a RRule instance, this triggers the static initializer of RRule, Recur, TemporalAdapter and then TemporalComparator (in that order), leaving TemporalAdapter.COMPARATOR as null, as TemporalComparator.INSTANCE is still null. 

**To Reproduce**
```
package net.fortuna.ical4j;

import net.fortuna.ical4j.model.Period;
import net.fortuna.ical4j.model.TemporalAdapter;
import net.fortuna.ical4j.model.component.VEvent;
import net.fortuna.ical4j.model.property.RRule;

import java.time.ZonedDateTime;
import java.time.temporal.Temporal;

public class Main {
    static {
//        TemporalAdapter.isBefore(ZonedDateTime.now(), ZonedDateTime.now());
    }

    public static void main(String[] args) {
        // this triggers the static initializer of RRUle, Recur, TemporalAdapter and then TemporalComparator
        // (in that order), leaving TemporalAdapter.COMPARATOR as null.
        new RRule<>("FREQ=WEEKLY;COUNT=3");


        Period<Temporal> period = new Period<>(ZonedDateTime.now(), ZonedDateTime.now());
        VEvent event = new VEvent(ZonedDateTime.now(), "event");

        // NPE here because TemporalAdapter.COMPARATOR is null
        event.calculateRecurrenceSet(period);
    }
}
```
Enabling the line of code in the static initializer is my current workaround.

**Expected behavior**
TemporalAdapter.COMPARATOR should not be null

**Screenshots**
Stack trace : 
```
Exception in thread "main" java.lang.RuntimeException: Error loading timezone
	at net.fortuna.ical4j.model.TimeZoneLoader.lambda$loadVTimeZone$1(TimeZoneLoader.java:85)
	at net.fortuna.ical4j.util.MapTimeZoneCache.getTimezone(MapTimeZoneCache.java:25)
	at net.fortuna.ical4j.model.TimeZoneLoader.loadVTimeZone(TimeZoneLoader.java:76)
	at net.fortuna.ical4j.model.DefaultZoneRulesProvider.provideRules(DefaultZoneRulesProvider.java:55)
	at java.base/java.time.zone.ZoneRulesProvider.getRules(ZoneRulesProvider.java:234)
	at java.base/java.time.ZoneRegion.ofId(ZoneRegion.java:120)
	at java.base/java.time.ZoneId.of(ZoneId.java:408)
	at java.base/java.time.ZoneId.of(ZoneId.java:356)
	at net.fortuna.ical4j.model.TimeZoneRegistry.lambda$getGlobalZoneId$1(TimeZoneRegistry.java:69)
	at java.base/java.util.Optional.map(Optional.java:265)
	at net.fortuna.ical4j.model.TimeZoneRegistry.getGlobalZoneId(TimeZoneRegistry.java:69)
	at net.fortuna.ical4j.model.parameter.TzId.toZoneId(TzId.java:85)
	at net.fortuna.ical4j.model.property.DateProperty.getDate(DateProperty.java:156)
	at net.fortuna.ical4j.model.Component.calculateRecurrenceSet(Component.java:353)
	at net.fortuna.ical4j.Main.main(Main.java:26)
Caused by: net.fortuna.ical4j.data.ParserException: Error at line 157:null
	at net.fortuna.ical4j.data.CalendarParserImpl.parse(CalendarParserImpl.java:167)
	at net.fortuna.ical4j.data.CalendarBuilder.build(CalendarBuilder.java:197)
	at net.fortuna.ical4j.data.CalendarBuilder.build(CalendarBuilder.java:185)
	at net.fortuna.ical4j.data.CalendarBuilder.build(CalendarBuilder.java:172)
	at net.fortuna.ical4j.model.TimeZoneLoader.lambda$loadVTimeZone$1(TimeZoneLoader.java:81)
	... 14 more
Caused by: java.lang.NullPointerException
	at net.fortuna.ical4j.model.TemporalAdapter.isBefore(TemporalAdapter.java:354)
	at net.fortuna.ical4j.model.component.Observance.getLatestOnset(Observance.java:170)
	at net.fortuna.ical4j.model.TimeZone.getRawOffset(TimeZone.java:208)
	at net.fortuna.ical4j.model.TimeZone.<init>(TimeZone.java:92)
	at net.fortuna.ical4j.model.TimeZone.<init>(TimeZone.java:80)
	at net.fortuna.ical4j.data.DefaultContentHandler.endComponent(DefaultContentHandler.java:125)
	at net.fortuna.ical4j.data.CalendarParserImpl$ComponentParser.parse(CalendarParserImpl.java:449)
	at net.fortuna.ical4j.data.CalendarParserImpl$PropertyListParser.parse(CalendarParserImpl.java:217)
	at net.fortuna.ical4j.data.CalendarParserImpl.parseCalendar(CalendarParserImpl.java:126)
	at net.fortuna.ical4j.data.CalendarParserImpl.parseCalendarList(CalendarParserImpl.java:188)
	at net.fortuna.ical4j.data.CalendarParserImpl.parse(CalendarParserImpl.java:161)
	... 18 more
```

**Environment (please complete the following information):**
 - OS: Linux debian 12.9
 - Java Version [e.g. 7, 8, 9]
 ❯ java -version
openjdk version "21.0.6" 2025-01-21 LTS
OpenJDK Runtime Environment Temurin-21.0.6+7 (build 21.0.6+7-LTS)
OpenJDK 64-Bit Server VM Temurin-21.0.6+7 (build 21.0.6+7-LTS, mixed mode, sharing)
 - iCal4j Version: 4.1.0 (`develop` branch, current commit bd0aba46 )

**Additional context**
Let me know if I should provide further information.

--- a/src/main/java/net/fortuna/ical4j/model/TemporalAdapter.java
+++ b/src/main/java/net/fortuna/ical4j/model/TemporalAdapter.java
@@ -34,8 +34,6 @@ import java.util.Objects;
  */
 public class TemporalAdapter<T extends Temporal> implements Serializable {
 
-    private static final TemporalComparator COMPARATOR = TemporalComparator.INSTANCE;
-
     /**
      * The iCalendar-compliant string representation of a {@link Temporal} value.
      */
@@ -345,11 +343,11 @@ public class TemporalAdapter<T extends Temporal> implements Serializable {
     }
 
     public static <T extends Temporal> boolean isBefore(T date1, T date2) {
-        return COMPARATOR.compare(date1, date2) < 0;
+        return TemporalComparator.INSTANCE.compare(date1, date2) < 0;
     }
 
     public static <T extends Temporal> boolean isAfter(T date1, T date2) {
-        return COMPARATOR.compare(date1, date2) > 0;
+        return TemporalComparator.INSTANCE.compare(date1, date2) > 0;
     }
 
     @Override

